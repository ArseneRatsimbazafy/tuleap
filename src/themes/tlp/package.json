{
  "author": "Enalean Team",
  "name": "@tuleap/tlp",
  "description": "Tuleap CSS framework",
  "license": "GPL-2.0-or-later",
  "main": "index.js",
  "types": "./src/index.d.ts",
  "private": true,
  "dependencies": {
    "@formatjs/intl-getcanonicallocales": "^1.4.3",
    "@formatjs/intl-numberformat": "^5.6.0",
    "@formatjs/intl-pluralrules": "^3.4.5",
    "@formatjs/intl-relativetimeformat": "^7.2.5",
    "@webcomponents/custom-elements": "^1.4.2",
    "flatpickr": "^4.6.6",
    "jquery": "^3.5.1",
    "popper.js": "^1.16.1",
    "select2": "^4.0.13"
  },
  "devDependencies": {
    "codemirror": "^5.57.0",
    "dompurify": "^2.2.0"
  },
  "config": {
    "bin": "../../../node_modules/.bin"
  },
  "scripts": {
    "test": "$npm_package_config_bin/jest"
  }
}
