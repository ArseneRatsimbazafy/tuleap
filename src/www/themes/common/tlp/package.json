{
  "name": "tlp",
  "version": "0.0.1",
  "description": "Tuleap CSS framework",
  "author": "Enalean",
  "license": "GPL-2.0",
  "main": "index.js",
  "private": true,
  "dependencies": {
    "core-js": "^3.2.0",
    "dom4": "^2.1.5",
    "flatpickr": "1.9.1",
    "jquery": "2.1.0",
    "popper.js": "^1.14.7",
    "regenerator-runtime": "^0.13.3",
    "select2": "^4.0.10",
    "whatwg-fetch": "^3.0.0"
  },
  "devDependencies": {
    "@types/select2": "^4.0.48",
    "codemirror": "^5.48.4",
    "dompurify": "^1.0.11"
  },
  "types": "./src/index.d.ts",
  "config": {
    "bin": "../../../../../node_modules/.bin"
  },
  "scripts": {
    "build": "$npm_package_config_bin/webpack --config webpack.prod.js",
    "watch": "$npm_package_config_bin/webpack --config webpack.dev.js --watch"
  }
}
