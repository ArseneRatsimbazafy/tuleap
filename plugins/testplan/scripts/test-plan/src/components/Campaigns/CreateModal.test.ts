/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import Vue from "vue";
import { shallowMount } from "@vue/test-utils";
import { createTestPlanLocalVue } from "../../helpers/local-vue-for-test";
import * as tlp from "tlp";
import CreateModal from "./CreateModal.vue";
import { Modal } from "tlp";

jest.mock("tlp", () => {
    return {
        __esModule: true,
        modal: jest.fn(),
    };
});

describe("CreateModal", () => {
    let local_vue: typeof Vue;

    beforeEach(async () => {
        local_vue = await createTestPlanLocalVue();
    });

    it("Display the modal when mounted", () => {
        const modal_show = jest.fn();
        jest.spyOn(tlp, "modal").mockImplementation(() => {
            return ({
                show: modal_show,
            } as unknown) as Modal;
        });

        const wrapper = shallowMount(CreateModal, {
            localVue: local_vue,
        });

        expect(modal_show).toHaveBeenCalledTimes(1);
        expect(wrapper.element).toMatchSnapshot();
    });
});
