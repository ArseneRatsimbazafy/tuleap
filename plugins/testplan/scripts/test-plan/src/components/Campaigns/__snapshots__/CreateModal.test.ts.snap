// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateModal Display the modal when mounted 1`] = `
<div
  aria-labelledby="test-plan-create-campaign-modal-title"
  class="tlp-modal"
  role="dialog"
>
  <div
    class="tlp-modal-header"
  >
    <h1
      class="tlp-modal-title"
      id="test-plan-create-campaign-modal-title"
    >
      <i
        aria-hidden="true"
        class="fa fa-plus tlp-modal-title-icon"
      />
       
      <translate-stub
        tag="span"
      >
        Create new campaign
      </translate-stub>
    </h1>
     
    <div
      aria-label="Close"
      class="tlp-modal-close"
      data-dismiss="modal"
      role="button"
      tabindex="0"
    >
      
            Ã—
        
    </div>
  </div>
   
  <div
    class="tlp-modal-body"
  >
    <div
      class="tlp-form-element"
    >
      <label
        class="tlp-label"
        for="new-campaign-label"
      >
        <translate-stub
          tag="span"
        >
          Campaign label
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fa fa-asterisk"
        />
      </label>
       
      <input
        class="tlp-input"
        id="new-campaign-label"
        type="text"
      />
    </div>
     
    <div
      class="tlp-form-element"
    >
      <label
        class="tlp-label"
        for="new-campaign-tests-selector"
      >
        <translate-stub
          tag="span"
        >
          Choose tests
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fa fa-asterisk"
        />
      </label>
       
      <select
        class="tlp-select"
        id="new-campaign-tests-selector"
      >
        <option
          data-current-language="en_US"
          data-msgid="No tests"
          value="none"
        >
          No tests
        </option>
         
        <option
          data-current-language="en_US"
          data-msgid="All tests"
          value="all"
        >
          All tests
        </option>
         
        <option
          data-current-language="en_US"
          data-msgid="
                    All tests in %{ milestone_title }
                "
          selected="selected"
          value="milestone"
        >
          
                    All tests in Milestone Title
                
        </option>
      </select>
    </div>
  </div>
   
  <div
    class="tlp-modal-footer"
  >
    <button
      class="tlp-button-primary tlp-button-outline tlp-modal-action"
      data-current-language="en_US"
      data-dismiss="modal"
      data-msgid="
            Close
        "
      type="button"
    >
      
            Close
        
    </button>
     
    <button
      class="tlp-button-primary tlp-modal-action"
      type="button"
    >
      <i
        aria-hidden="true"
        class="fa fa-save tlp-button-icon"
      />
       
      <translate-stub
        tag="span"
      >
        Create campaign
      </translate-stub>
    </button>
  </div>
</div>
`;
